subprojects/normalization-java/src/main/java/org/gradle/internal/normalization/java/impl/AsmConstants.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.internal.normalization.java.impl;
+
+import org.objectweb.asm.Opcodes;
+
+interface AsmConstants {
+    int ASM_LEVEL = Opcodes.ASM7;
```

Comment:
Does it make sense to have this class at all? I think we should just inline `ASM7` everywhere. It's not like we need to keep this the same for every visitor we have or something.

Reply:
I think it makes sense to the same `ASM_LEVEL` uniformly on the Gradle code-base, given that we only use one version of asm. We don't need to use this constant, though.

Maybe the interface was mostly there to suppress the deprecation warning introduced by `ASM7_EXPERIMENTAL` and we can inline the class again.

Reply:
I inlined it in this package. I kept the other `AsmConstants` references from all over the codebase for now.